inline.h

```diff
@@ -57,6 +57,94 @@ Perl_av_count(pTHX_ AV *av)
     return AvFILL(av) + 1;
 }
 
+/* ------------------------------- av.c ------------------------------- */
+
+/*
+=for apidoc av_store_simple
+
+This is a cut-down version of av_store that assumes that the array is
+very straightforward - no magic, not readonly, and AvREAL - and that
+C<key> is not negative. This function MUST NOT be used in situations
+where any of those assumptions may not hold.
+
+Stores an SV in an array.  The array index is specified as C<key>.  The
+return value will be C<NULL> if the operation failed or if the value did not
+need to be actually stored within the array (as in the case of tied
+arrays).  Otherwise, it can be dereferenced
```

Comment:
I don't think there's actually any way for this to return NULL - all possible failures look to be fatal, and it can't be a tied array - so a chunk of this documentation could be simplified.

Reply:
Yes, agreed. Too much copy and paste from `av_store` left in.

Reply:
Update pushed.